sympy/sets/handlers/intersection.py

```diff
@@ -264,7 +264,7 @@ def intersection_sets(self, other): # noqa:F811
             # not helpful here: multiple parametric solutions.
             if len(solns) == 0:
                 return EmptySet
-            elif any(not isinstance(s, int) and s.free_symbols
+            elif any(s.free_symbols and not isinstance(s, int)
```

Comment:
It is possible for `s` to be an `int` here? If it is then `s.free_symbols` would give an attribute error.

My guess is that this is a workaround for a bug that resulted in unsympified types being returned from `diophantine`. I think the `not isinstance(s, int)` should be removed if the tests pass with this change. If there is a failure then I guess that's a bug in diophantine. It should be fixed rather than worked around.

Reply:
You are probably correct. I realized that these changes were not always good when I found the `x.is_Integer and ...`, but forgot about this one. Will not fix diophantine though. :-)

Reply:
Seems like it passes. Should we assume that the bug is fixed or that a proper test case is missing?

Reply:
Or rather the diophantine tests passed. Better wait for the other ones as well...

Reply:
Something like this is designed to hide bugs. I'd rather remove it so that the bugs can be seen and fixed even if that has to happen in future rather than now.